tests/src/python/test_qgsblockingnetworkrequest.py

```diff
@@ -70,7 +70,7 @@ def testFetchBadUrl2(self):
         self.assertIn('Not Found', request.errorMessage())
         reply = request.reply()
         self.assertEqual(reply.error(), QNetworkReply.ContentNotFoundError)
-        self.assertFalse(reply.content())
+        self.assertEqual(reply.content(), '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"\n        "http://www.w3.org/TR/html4/strict.dtd">\n<html>\n    <head>\n        <meta http-equiv="Content-Type" content="text/html;charset=utf-8">\n        <title>Error response</title>\n    </head>\n    <body>\n        <h1>Error response</h1>\n        <p>Error code: 404</p>\n        <p>Message: File not found.</p>\n        <p>Error code explanation: HTTPStatus.NOT_FOUND - Nothing matches the given URI.</p>\n    </body>\n</html>\n')
```

Comment:
In #42630 this test was failing and it actually should've failed when we merged #42553 .. Don't know why it passed and let's see if it fails now..

Reply:
Turned out some of the local server stuff was refactored and using a mocked server now. Needed to provide that with a response body, works now.